OsmAnd/src/net/osmand/plus/GPXDatabase.java

```diff
@@ -258,9 +258,47 @@ SQLiteConnection openConnection(boolean readonly) {
 				onUpgrade(conn, version, DB_VERSION);
 			}
 		}
+		cleanUpDatabase(conn);
```

Comment:
It is too often. Please move cleanUpDatabase to constructor (239 line)

Reply:
Done